src/content/articles/why-typescript-doesnt-include-a-throws-keyword.mdx

```diff
@@ -0,0 +1,374 @@
+
+JavaScript, on the other hand, is a much more flexible language than many traditional strongly-typed languages.
+JavaScript and TypeScript include several features that make "happy" and "unhappy" path management easier, including the ones described later in [Preferred Alternatives](#preferred-alternatives):
+
+- [First-class functions](#first-class-functions): providing inline functions that can be provided multiple parameters
+- [Union types](#union-types): allowing returned values to match one of several possible shapes
+
+Nowadays, many JavaScript and TypeScript developers prefer using those languages' flexible features to avoid throwing exceptions.
+In doing so, they've lessened the frequency with which their code throws exceptions, lessening the need for `throw` types or checked exceptions.
+
+### Type System Complexity
+
+Every addition to the TypeScript language increases the complexity of its type system.
+Throw types would need to also be factored into the type checker's assignability checks for function types.
+However, being able to define throw types that don't excessively report on valid code is tricky.
+
+Take the case of Object getters and setters potentially throwing errors.
+It would be very inconvenient for developers if setting the `length` property of an Array always necessitated adding a throws type.
+But, TypeScript doesn't have the ability to represent integer types more precise than `number`.
+
+Additional tricky type system questions include:
+
+- How should interface and object type properties indicate they may throw errors?
+- If code isn't annotated as throwing an exception, should adding a `try` block around it still be allowed?
+- How should type annotations indicate that a function's parameter may be a function that can throw errors, but those won't be raised to calling code? (e.g. [`setTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout))
+
+TypeScript adding throws types would necessitate developers learning those answers in order to effectively write type-safe functions with throw types.
+Even if the answers are straightforward, that's still added complexity to what developers need to understand to write TypeScript code.
+
+## Preferred Alternatives
+
+When working in a typed language such as TypeScript, it's useful to design code in ways that can be modeled by the language's type system.
+Doing so allows the type system to better understand the code and provide more assistance to developers using it.
+
+### First-Class Functions
+
+JavaScript is known in part for its support for "first-class functions": meaning new functions may be provided as values for function arguments and variables.
+Many APIs developed in JavaScript chose to use first-class functions instead of throwing exceptions.
+
+For example, the [Node.js `fs.readFile` API](https://nodejs.org/api/fs.html) designed before JavaScript promises have developers provide a function to be called on completion.
```

Comment:
While you mention promises: the type for `Promise` like `Promise<string>` should likely be changed (or augmented) to contain the error, like `Promise<TheErrorType, string>`. Maybe by like re-defining `Promise<T>` as `PromiseWithError<any, T>`, or having a value like `void` indicating that the Promise won't throw an error, as far as we know).
This would similarly be a hard change to make for the ecosystem.

Reply:
Ha, yeah, I think that's a can of worms I'm going to try to avoid here. Agreed that for similar reasons stated separately, it's ... a lot.